win32ss/printing/base/winspool/printers.c

```diff
@@ -449,9 +449,17 @@ EnumPrintersA(DWORD Flags, PSTR Name, DWORD Level, PBYTE pPrinterEnum, DWORD cbB
  
     /* Ref: https://stackoverflow.com/questions/41147180/why-enumprintersa-and-enumprintersw-request-the-same-amount-of-memory */
     bReturnValue = EnumPrintersW(Flags, pwszName, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
-    HeapFree(hProcessHeap, 0, pwszName);
+    TRACE("*pcReturned is '%d' and bReturnValue is '%d' and GetLastError is '%ld'.\n",
+          *pcReturned, bReturnValue, GetLastError());
 
-    TRACE("*pcReturned is '%d' and bReturnValue is '%d' and GetLastError is '%ld'.\n", *pcReturned, bReturnValue, GetLastError());
+    if (pwszName)
+        HeapFree(hProcessHeap, 0, pwszName);
+
+    if (!bReturnValue)
```

Comment:
Good catch, thanks!

Reply:
This call is expected to fail with a GetLastError of ERROR_INSUFFICIENT_BUFFER when cbBuf = 0 when we probe to see how large the buffer should be. Should this condition be checked and the failure message be suppressed in this case?

Reply:
> failure message be suppressed

I agree: it should simply be removed actually.